modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -74,18 +75,26 @@ def fetch_cached_or_service(validated_params)
             start_date: [validated_params[:start_date], one_year_ago].min,
             end_date: [validated_params[:end_date], one_year_from_now].max
           )
+
           Mobile::V0::Appointment.set_cached(@current_user, appointments)
           Rails.logger.info('mobile appointments service fetch', user_uuid: @current_user.uuid)
         end
 
+        unless validated_params[:included]&.include?('pending')
+          appointments.filter! { |appt| appt.status.in?(%w[COMMUNITY_CARE_REQUEST VA_REQUEST]) }
+        end
         appointments.reverse! if validated_params[:reverse_sort]
+
         appointments
       end
 
       def paginate(appointments, validated_params)
         appointments = appointments.filter do |appointment|
-          appointment.start_date_utc.between? validated_params[:start_date], validated_params[:end_date]
+          appointment.start_date_utc.between?(
+            validated_params[:start_date].beginning_of_day, validated_params[:end_date].end_of_day
```

Comment:
I changed these because at one point I was testing with a very limited date range and experiencing issues.